stdlib/REPL/src/precompile.jl

```diff
@@ -37,9 +37,22 @@ function repl_workload()
     UP_ARROW = "\e[A"
     DOWN_ARROW = "\e[B"
 
+    atreplinit() do repl
+        # Main is closed so we can't evaluate in it
+        # but atreplinit runs at a time that repl.mistate === nothing
+        # so REPL.activate fails. So do it async, with an initial
+        # sleep on the repl instructions
+        t = @async begin
+            while repl.mistate === nothing
+                yield()
+            end
+            REPL.activate(REPL.Precompile; interactive_utils=false)
+        end
+        Base.errormonitor(t)
+    end
+
     repl_script = """
-    import REPL
-    REPL.activate(REPL.Precompile; interactive_utils=false) # Main is closed so we can't evaluate in it
+    sleep(1) # give the above async atreplinit a chance to change active_module
```

Comment:
Sigh, I thought #55316 was supposed to fix this, but this API still feels like a disaster there.

I think you can replace this with an `Event` wait signaled by the atreplinit (and move the rest of the contents of the async block here?)